doc/BUILD.rst

```diff
@@ -0,0 +1,730 @@
+
+ANGLE
+"""""
+
+Shader Transpiler (``-Dshader_transpiler``)
+'''''''''''''''''''''''''''''''''''''''''''
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+For using OpenGL ES, the shader transpiler is necessary for converting Taisei's
+shaders to a format usable by that driver.
+
+.. code:: sh
+
+   meson configure build/ -Dshader_transpiler=true
+
+ANGLE Library Paths (``-Dangle_lib*``)
+''''''''''''''''''''''''''''''''''''''
+
+* Default: ``(null)``
+* Options: ``/path/to/libGLESv2.{dll,dylib,so}``/``path/to/libEGL.{dll,dylib,so}``
+
+``-Dangle_libgles`` and ``-Dangle_libegl`` provide the full paths to the ANGLE
+libraries necessary for that engine.
+
+Generally, both need to be supplied at the same time.
+
+.. code:: sh
+
+   # for Linux
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.dylib -Dangle_libegl=/path/to/libEGL.dylib
+   # for macOS
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.so -Dangle_libegl=/path/to/libEGL.so
+   # for Windows
+   meson configure build/ -Dangle_libgles=/path/to/libGLESv2.dll -Dangle_libegl=/path/to/libEGL.dll
+
+Install ANGLE (``-Dinstall_angle``)
+'''''''''''''''''''''''''''''''''''
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+Installs the ANGLE libraries supplied above through ``-Dangle_lib*``.
+
+Generally recommended when packaging ANGLE for distribution.
+
+.. code:: sh
+
+   meson configure build/ -Dinstall_angle=true
+
+Building ANGLE (Optional)
+'''''''''''''''''''''''''
+`ANGLE <https://github.com/google/angle>`__ is Google's graphics translation
+layer, intended for for Chromium. Taisei packages it with Windows builds to
+workaround some bugs and performance issues with many Windows OpenGL drivers,
+and it can be optionally packaged as as an experimental Metal renderer for
+macOS.
+
+You need to read
+`this guide <https://github.com/google/angle/blob/master/doc/DevSetup.md>`__ and
+set up Google's custom build system to get things going. However, the below
+commands might help you compiling what you need from it when you have that all
+set up.
+
+.. code:: sh
+
+   cd angle
+   python ./scripts/bootstrap.py
+   gclient sync
+   gn gen out/x64 --args='is_debug=false dcheck_always_on=false target_cpu="x64"'
+   ninja -C out/x64 libEGL libGLESv2
+
+It will output two files to ``angle/out/x64``:
+
+* ``libEGL.(*)``
+* ``libGLESv2.(*)``
+
+The file extension can be ``.dll`` for Windows, ``.dylib`` for macOS,
+and ``.so`` for Linux.
+
+Using ``-Dinstall_angle`` and ``-Dangle_lib*`` (see above), ``meson`` will copy
+those files over into the package itself when running the packaging steps.
+
+Packaging
+---------
+
+Archive Types
+"""""""""""""
+
+Taisei supports a wide variety of packaging and archive types, including
+``.zip``, ``.tar.gz``, ``.dmg`` (for macOS), ``.exe`` (for Windows), among
+others.
+
+Examples
+""""""""
+
+Linux
+'''''
+
+Compiling on Linux for Linux is fairly straightforward. We have ``meson``
```

Comment:
Again, huuuuuge asterisk here because of the SDL2 nuances…

Honestly, I don't think it's worth it documenting the release/packaging options in this file. This is a fairly niche topic, and you'd really need to get into the weeds to get everything factually correct. Maybe split it off into a `RELEASE.rst` for later, and leave this one dedicated to dependencies and build options.